source/common/common/utility.cc

```diff
@@ -67,6 +69,55 @@ bool StringUtil::atoul(const char* str, uint64_t& out, int base) {
   }
 }
 
+absl::string_view StringUtil::ltrim(absl::string_view source) {
+  source.remove_prefix(std::min(source.find_first_not_of(" \t\f\v\n\r"), source.size()));
+  return source;
+}
+
+absl::string_view StringUtil::rtrim(absl::string_view source) {
+  source.remove_suffix(source.size() - source.find_last_not_of(" \t\f\v\n\r") - 1);
+  return source;
+}
+
+absl::string_view StringUtil::trim(absl::string_view source) { return ltrim(rtrim(source)); }
+
+bool StringUtil::find(absl::string_view source, absl::string_view delimiters,
+                      absl::string_view key_token, bool trim_whitespace) {
+  const std::vector<absl::string_view> tokens = splitToken(source, delimiters, !trim_whitespace);
+  if (tokens.empty()) {
+    return key_token.empty();
+  }
+
+  if (trim_whitespace) {
+    for (auto token : tokens) {
+      if (key_token == trim(token)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  return std::find(tokens.begin(), tokens.end(), key_token) != tokens.end();
+}
+
+absl::string_view StringUtil::cropRight(absl::string_view source, absl::string_view delimiter,
+                                        bool trim_whitespace) {
+  const absl::string_view::size_type pos = source.find(delimiter);
+  if (pos != absl::string_view::npos) {
+    source.remove_suffix(source.size() - pos);
+  }
+  return trim_whitespace ? trim(source) : source;
```

Comment:
maybe just right-trim the source, since you are just cropping right?  either way is fine.

Reply:
Rtrim seems more correct actually 👍